### PR TITLE
Add Travis build

### PR DESCRIPTION
**Pull request by palfrey**
One of the build steps (just one I've found useful) checks that the minimum versions specified in the Cargo.toml can actually be used to build the program. As with most cases when I try this, it turns out they don't work, so I've fixed those as well.

https://travis-ci.org/github/palfrey/firestarter/builds/728715530 demos this working

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+language: rust
+
+rust:
+ - 1.31.0
```
Comment (palfrey) [inline .travis.yml:4]: 1.31.0 was picked as it's the minimum needed for Rust 2018, and that seems to be enough

File: README.md [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 [![crates.io](https://img.shields.io/crates/v/firestarter.svg)](https://crates.io/crates/firestarter)
 [![Patreon](https://img.shields.io/badge/patreon-become%20a%20patron-red.svg)](https://www.patreon.com/mopemope)
+[![Build Status](https://travis-ci.org/palfrey/firestarter.svg?branch=master)](https://travis-ci.org/palfrey/firestarter)
```
Comment (palfrey) [inline README.md:5]: This should be corrected to your own Travis build

**Comment by mopemope** [comment]
@palfrey Sorry, I've merged the latest source into master, please check again.

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+language: rust
+
+rust:
+ - 1.36.0
```
Comment (palfrey) [inline .travis.yml:4]: 1.36.0 needed by anyhow and nix

**Comment by palfrey** [comment]
> @palfrey Sorry, I've merged the latest source into master, please check again.

Merged master, fixed the issues that I found as a result.

**Comment by mopemope** [comment]
@palfrey Merged. Thanks !

**Pull request merged**